libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2713,6 +2713,8 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 				annotation.isPure = _memberAccess.expression().annotation().isPure;
 		}
 	}
+	else if (exprType->category() == Type::Category::Module)
+		annotation.isPure = _memberAccess.expression().annotation().isPure;
```

Comment:
What do you think about adding a test for direct `A.B.L.foo()` access (where `A` and `B` are modules) which would exercise this path?